### PR TITLE
[Queues] Clarify waitUntil interaction

### PR DESCRIPTION
**Pull request by elithrar**

### CODE REVIEW

File: content/queues/platform/javascript-apis.md [hunk 1]
```diff
@@ -131,6 +129,20 @@ addEventListener('queue', (event) => {
 ```
 
 In service worker syntax, `event` provides the same fields and methods as `MessageBatch`, as defined below, in addition to [`waitUntil()`](https://developer.mozilla.org/en-US/docs/Web/API/ExtendableEvent/waitUntil).
+  
+#### Using ctx.waitUntil()
+  
+Using [`ctx.waitUntil()`](/workers/runtime-apis/fetch-event/#waituntil) extends the lifetime of the "queue" event handler. It accepts a Promise-based task which the Workers runtime will execute before the handler terminates but without blocking the response.
```
Comment (a-robinson) [inline content/queues/platform/javascript-apis.md]: > but without blocking the response.

This isn't true as currently implemented -- promises passed to `waitUntil` *do* currently block the response in `queue()` event handlers. This differs from `fetch()` handlers but matches the behavior of `scheduled()` handlers.

As discussed before I'd like to change it to behave more like `fetch`, but that'll require a compatibility flag and until then we should document how it actually works :)

**Review by a-robinson — CHANGES_REQUESTED** [review]

File: content/queues/platform/javascript-apis.md [hunk 2]
```diff
@@ -131,6 +129,20 @@ addEventListener('queue', (event) => {
...
+  
+{{<Aside type="warn">}}
+  
+We do not generally recommend using `ctx.waitUntil()` to process message batches directly, as any failure within the context of `waitUntil` can't be caught by your own code.
```
Comment (deadlypants1973) [inline content/queues/platform/javascript-apis.md]:
```suggestion
We do not generally recommend using `ctx.waitUntil()` to process message batches directly, as any failure within the context of `waitUntil` cannot be caught by your own code.
```

File: content/queues/platform/javascript-apis.md (hunk 1 above)
Comment (deadlypants1973) [inline content/queues/platform/javascript-apis.md]:
```suggestion
Using [`ctx.waitUntil()`](/workers/runtime-apis/fetch-event/#waituntil) extends the lifetime of the queue event handler. It accepts a Promise-based task which the Workers runtime will execute before the handler terminates but without blocking the response.
```

Adding this on top of Alex's comment but queue should either be in `monospaces` or left without "". We avoid "" in docs unless we are `"quoting to point to a string in code"`.

Comment (elithrar) [reply content/queues/platform/javascript-apis.md]: (Fixed this — it should have said blocking. Bad copy + paste)

Comment (elithrar) [reply content/queues/platform/javascript-apis.md]: Copied from https://developers.cloudflare.com/workers/runtime-apis/fetch-event/#waituntil so should fix there 😄

**Review by a-robinson — APPROVED** [review]

**Pull request closed without merging**